### PR TITLE
Try except

### PR DESCRIPTION
**Pull request by Ovec8hkin**
Refactored to remove unnecessary try-except blocks in favor of an easier to read and maintain set of conditional statements for generation of the ssaraopt string in `download_ssara_rsmas.py`.

Solves issue #71 

We may want to consider moving this function and make a it a method of the `Template` object since it deals directly with the template options.

### CODE REVIEW

**Comment by gravelcycles** [comment]
This PR has 10 commits. I think only the last 3 mattered. I would recommend creating a new branch (branching from `master`) and then `git cherry-pick`

https://stackoverflow.com/questions/9339429/what-does-cherry-picking-a-commit-with-git-mean

**Comment by Ovec8hkin** [comment]
Yah, not sure why those are there, as they have been merged already. Just review as is.

File: download_ssara_rsmas.py [hunk 1]
```diff
@@ -72,35 +72,39 @@ def generate_ssaraopt_string(template_file):
...
-    # add parallel doenload option. If ssaraopt.parallelDownload not given use default value
-    try:
-       parallelDownload = template_options['ssaraopt.parallelDownload']
-    except:
-       parallelDownload = '30'     # default
-    ssaraopt=ssaraopt+' --parallel='+parallelDownload
+
+    ssaraopt_keys = ['ssaraopt.platform', 'ssaraopt.relativeOrbit', 'ssaraopt.frame', 'ssaraopt']
+
+    bad_key=False
```
Comment (gravelcycles) [inline download_ssara_rsmas.py]: How about this?
```python
ssaraopt_keys = ['ssaraopt.platform', 'ssaraopt.relativeOrbit', 'ssaraopt.frame']

bad_key = False
for key in ssaraopt_keys:
    if key not in template_options:
        bad_key = True
        if 'ssaraopt' not in template_options:
            raise Exception('no ssaraopt or ssaraopt.platform, relativeOrbit, frame found')
        else:
            ssaraopt = template_options['ssaraopt']
            break

if not bad_key:
    platform = template_options['ssaraopt.platform']
    relativeOrbit = template_options['ssaraopt.relativeOrbit']
    frame = template_options['ssaraopt.frame']
    ssaraopt='--platform={} --relativeOrbit={} --frame={}'.format(platform, relativeOrbit, frame)
    ...
```

Comment (gravelcycles) [reply download_ssara_rsmas.py]: Seems a bit cleaner to deal with the `ssaraopt` key separately

File: download_ssara_rsmas.py [hunk 2]
```diff
@@ -72,35 +72,39 @@ def generate_ssaraopt_string(template_file):
...
+    for key in ssaraopt_keys:
```
Comment (gravelcycles) [inline download_ssara_rsmas.py:82]: Can you add a comment explaining this block of logic?

File: download_ssara_rsmas.py (hunk 1 above)
Comment (gravelcycles) [inline download_ssara_rsmas.py]: Nit: add a space - `bad_key = True`

**Review by gravelcycles — CHANGES_REQUESTED** [review]
Think about this slight change. I think the logic flow is a bit more comprehendible this way.

File: dataset_template.py [hunk 3]
```diff
@@ -23,15 +24,18 @@ def __init__(self, custom_template_file):
 
             :param custom_template_file: file, the template file to be accessed
         """ 
+
+        # Joshua Zahner - 02/2019
```
Comment (Ovec8hkin) [inline dataset_template.py:28]: @2gotgrossman Could you provide your opinion on the following code block? I think this is creating a copy of the template file in a new directory, but I don't think that should be a part of this object, as it doesn't deal with the data, but rather file handling. Thoughts?

File: download_ssara_rsmas.py (hunk 1 above)
Comment (Ovec8hkin) [reply download_ssara_rsmas.py]: Your right, I was trying to come up with a not bad way to do that, but missed this easy fix.

File: dataset_template.py (hunk 3 above)
Comment (gravelcycles) [reply dataset_template.py:28]: Yes I can, but could you please move it to a separate PR?

**Pull request closed without merging**